guidelines/coding-guidelines.md

```diff
@@ -0,0 +1,6 @@
+## Performance related
+
+* Avoid allocations in hot paths:
+* Avoid LINQ.
```

Comment:
"LINQ" is an overloaded term. Does it mean `System.Linq`, declarative LINQ syntax,or something else? What is wrong with using it? Are any parts of `System.Linq` OK?